### DISCUSSION TITLE
Why is "Name" suddenly a required field when updating a product by SKU?

### DISCUSSION THREAD

**Discussion by Taegost** [Sellbrite/Sellbrite-API#25] [Q&A]
At some point between October 5th, 2023 and today (November 8th, 2023), the "Name" field suddenly became required when doing a PUT to the /products/{sku} endpoint. Does anybody know why this is? It just broke all the automation my business relies on to do updates for new products and is going to require me to double the number of API calls because now I have to first get the name of the product, *THEN* do my PUT to update it...
Here's the documentation showing the change, too: https://developer.sellbrite.com/endpoints/products-sku-post

**Comment by Taegost** [comment]
Now that I've got one of my automations back on track, I found that the "name" field being required is inconsistent. If all I'm doing is updating the Product URL, I have to put a name, unless I make a GET call to the endpoint first to get the name, then I don't have to include it in the PUT call. Also, if I'm updating a bunch of fields at once, I don't have to enter the name either, even without a GET call immediately before.

**Comment by vmishra-godaddy** [comment]
@Taegost Would you mind posting the request/response(excluding API key) for both API calls?

**Reply by Taegost** [reply]
I don't have the exact json that was used as the request in the first scenario, but here are the details I do have for the first failure that prompted this discussion (I'd also like to point out that this workflow worked perfectly fine on October 5th, nothing was changed between then and when it failed on November 8th).

**Request:**
PUT to endpoint `https://api.sellbrite.com/v1/products/DM-32-01-cazarions-grave`
The Body only contained a single item, an update to the `store_product_url` with the value of `https://diceninjagaming.com/products/cazarions-grave`

**Response:**
>{"status":"rejected","reason":{"message":"400 - \"{\\\"error\\\":\\\"ProductSingle: Validation failed: Name cannot be blank\\\"}\"","name":"Error","stack":"Error: 400 - \"{\\\"error\\\":\\\"ProductSingle: Validation failed: Name cannot be blank\\\"}\"\n    at createError (/usr/local/lib/node_modules/n8n/node_modules/axios/lib/core/createError.js:16:15)\n    at settle (/usr/local/lib/node_modules/n8n/node_modules/axios/lib/core/settle.js:17:12)\n    at RedirectableRequest.handleResponse (/usr/local/lib/node_modules/n8n/node_modules/axios/lib/adapters/http.js:238:9)\n    at RedirectableRequest.emit (node:events:529:35)\n    at RedirectableRequest.emit (node:domain:489:12)\n    at RedirectableRequest._processResponse (/usr/local/lib/node_modules/n8n/node_modules/follow-redirects/index.js:368:10)\n    at ClientRequest.RedirectableRequest._onNativeResponse (/usr/local/lib/node_modules/n8n/node_modules/follow-redirects/index.js:65:10)\n    at Object.onceWrapper (node:events:632:26)\n    at ClientRequest.emit (node:events:517:28)\n    at ClientRequest.emit (node:domain:489:12)"}}

The error stopped if a did a GET to the same endpoint right before doing the PUT.

Later on in the same workflow, I do call another PUT to update more fields, and it never required the Name field to be updated. Again, I don't have the exact request that was sent, but I do have the JSON body this time.
Here is the sample JSON body that was passed as a `PUT` to `https://api.sellbrite.com/v1/products/WF-32-01-fairy-xmas-tree` and did not require the Name parameter:

`{
  "sku": "WF-32-01-fairy-xmas-tree",
  "brand": "World Forge Minis",
  "manufacturer": "Dice Ninja Gaming",
  "manufacturer_model_number": "WF-32-01-fairy-xmas-tree",
  "description": "Dimensions: 1.9\"x1.9\"x2.75\"\n\nEmbrace the holiday spirit in your tabletop gaming adventures with our exquisite \"Christmas Tree\" miniatures. These 3D-printed marvels come in two enchanting variations: straight and curved, offering you the perfect choice to adorn your gaming world.\n\nWhether you opt for the straight, stately tree or the gracefully curved version, both are meticulously decorated with an array of festive streamers and delicate ornaments. These tiny treasures capture the essence of a Fairy Christmas, adding a touch of magic to your gaming tableau.\n\nImagine your characters embarking on quests, their journeys unfolding beneath the watchful branches of these beautifully adorned Christmas trees. As the streamers sway gently and the delicate balls glisten in the imaginary moonlight, you'll create an atmosphere that immerses your fellow players in the holiday wonder of your campaign.\n\nThese \"Christmas Tree\" miniatures aren't just game pieces; they're invitations to infuse your gaming sessions with the joy and enchantment of the season. With their elegant design and attention to detail, they are sure to instill curiosity and delight in both players and characters alike.\n\nChoose the \"Christmas Tree\" miniature that suits your gaming world, and let the magic of Fairy Christmas come alive at your gaming table. Elevate your campaign and make it an unforgettable journey through a world adorned with the spirit of the season. Don't miss out on the opportunity to add these enchanting miniatures to your collection today.",
  "package_unit_of_weight": "pounds",
  "image_list": "https://images.sellbrite.com/production/331626/WF-32-00-fairy-xmas-tree/3bc3d487-9318-5f14-9fda-27c56cda17ed.jpg",
  "store_product_url": "https://diceninjagaming.com/products/fairy-xmas-christmas-tree",
  "custom_attributes": {
    "dndsizecategory": "Large",
    "bundlename": "A Fairy Christmas",
    "releasemonth": "December",
    "releaseyear": "2022"
  }
}`
